src/Analyzer/Passes/RemoveUnusedProjectionColumnsPass.cpp

```diff
@@ -154,6 +154,28 @@ void RemoveUnusedProjectionColumnsPass::run(QueryTreeNodePtr & query_tree_node,
                 if (query_node->isDistinct())
                     continue;
             }
+            else
+            {
+                auto * union_node = query_or_union_node->as<UnionNode>();
+                chassert(union_node != nullptr);
+
+                /// We can't remove unused projections in the case of EXCEPT and INTERSECT
+                /// because it can lead to incorrect query results. Example:
```

Comment:
Could you pls also clarify what and where goes wrong in this case? It's hard to guess how smth might lead to a wrong result when it is "unused".